source/agora/network/Manager.d

```diff
@@ -1044,6 +1056,26 @@ public class NetworkManager
         return headers_updated;
     }
 
+    /// Retrieve any missing enrollments
+    public void getEnrollments (Height height, EnrollmentManager enroll_man,
+        scope UTXOFinder finder) @safe nothrow
+    {
+        Hash[] enroll_keys;
+        enroll_man.enroll_pool.getEnrollments(height).
+            each!(enroll => enroll_keys ~= enroll.utxo_key);
+
+        foreach (peer; this.peers)
+        {
+            auto peer_enrolls = peer.client.getEnrollments(height, enroll_keys.length);
+            foreach (enroll ; peer_enrolls)
+            {
+                if (!enroll_keys.canFind(enroll.utxo_key) && 
+                    enroll_man.addEnrollment(enroll, height, finder))
+                    enroll_keys ~= enroll.utxo_key;
+            }
+        }
```

Comment:
So this code:
1) Get a (newly allocated) array of all the enrollments;
2) Allocate a new array of the same size with just the `utxo_key`;
3) Queries *all peers* (no matter what), and for each peer:
4) Get all their enrollments (but only if they have more enrollments... If they have less, but a different set, no luck);
5) Add each enrollment, after doing a linear search in `enroll_keys`;

Not only is this approach extremely wasteful, as the pool checks if there's already an enrollment for this UTXO, hence the `getEnrollments` => `each` / `canFind` are superfluous, but it also won't work if, for example, we have 5 enrollments, and the other node has 4 **totally different** Enrollments.